praxis-isncsci/ui
# Use icon from SVG sprites

Closes #58 

## Problem

As I try to use the icon element in icon button, I realize the initial approach is not very efficient and makes it hard to use different icon themes. I need to improve the approach to have the icons stored in less assets and make it possible to switch icons on the app along with the CSS themes.

## Solution

1. I replaced the icon web components with a single SVG file that will contain our entire icon collection. This will allow us to easily load new icon collections to match other themes.
2. Inside the icon element, I use the `<use>` tag to load the right sprite.
3. I had to update the StoryBook config to ensure it moves the SVG icon file so it can be referenced from the stories.

## Testing

- [x] 1. `<praxis-isncsci-icon>` loads the graphics from SVG sprite sheet

<details>
  <summary>Test case</summary>

  1. Navigate to the story at `?path=/docs/webcomponents-praxisisncsciicon--docs`
  2. Confirm the icons load

</details>
